os/fs/driver/mtd/smart.c

```diff
@@ -5596,26 +5599,31 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 	
 	fvdbg("recovery data journal_seq : %d!!\n", dev->journal_seq);
 
-	/* Before calculate journal, check the journal contents first */
+	/* Before calculating Journal CRC, check the journal contents first
+	   Validity of physical sector number associated with journal
+	 */
 	if (psector >= (dev->geo.neraseblocks * dev->sectorsPerBlk)) {
 		fdbg("invalid psector : %d\n", psector);
 		goto error_with_checkin;
 	}
 
+	/* Verify that the sequence number stored in the log matches the sequence number from the loop in journal_scan */
 	if (UINT8TOUINT16(log->seq) != dev->journal_seq) {
 		fdbg("journal sequence is not match log->seq : %d dev->journal_seq : %d\n", UINT8TOUINT16(log->seq), dev->journal_seq);
 		goto error_with_checkin;
 	}
 
+	/* Validate the type of the journal entry */
 	type = GET_JOURNAL_TYPE(log->status);
-
 	if ((type == 0) || (type > SMART_JOURNAL_TYPE_ERASE)) {
 		fdbg("invalid type : %d\n", type);
 		goto error_with_checkin;
 
 	}
 	
-	/* Recovery step is Check crc -> Check something more based on type -> checkout -> verify based on type */
+	/* Recovery steps include: -
```

Comment:
1. check crc...
2.  subsequent...
3. checkout....
4. verification
This style will be better.

Reply:
Updated